portfolio/src/test/java/com/google/sps/CalendarGroupTest.java

```diff
@@ -0,0 +1,317 @@
+   * free time range.
+   */
+  @Test
+  public void allFreeTime() {
+    // Events:
+    // Possible: |---------------------------|
+    // Free:     |---------------------------|
+
+    List<CalendarEvent> events = new ArrayList<CalendarEvent>();
+    Instant startTime = Instant.now();
+    Instant endTime = startTime.plusSeconds(10000);
+    TimeRange targetFreeTime = TimeRange.fromStartEnd(startTime, endTime);
+
+    CalendarGroup eventGroup = new CalendarGroup(events, startTime, endTime);
+    List<TimeRange> freeTimeRanges = eventGroup.getFreeTimeRanges();
+
+    Assert.assertTrue(freeTimeRanges.get(0).equals(targetFreeTime));
+  }
+
+  /** Tests deleting and modifying a free time range. */
+  @Test
+  public void deleteAndModifyTimeRange() {
+    // Events:       |--One--|   |--Two--|
+    // Possible: |---------------------------|
+    // Free:     |---|       |---|       |---|
+    List<CalendarEvent> events = new ArrayList<CalendarEvent>();
+
+    Instant eventOneStart = Instant.now();
+    Instant eventOneEnd = eventOneStart.plusSeconds(1000);
+    Instant eventTwoStart = eventOneEnd.plusSeconds(1000);
+    Instant eventTwoEnd = eventTwoStart.plusSeconds(1000);
+
+    Instant startTime = eventOneStart.minusSeconds(1000);
+    Instant endTime = eventTwoEnd.plusSeconds(1000);
+
+    CalendarEvent eventOne = new CalendarEvent("event one", eventOneStart, eventOneEnd);
+    CalendarEvent eventTwo = new CalendarEvent("event two", eventTwoStart, eventTwoEnd);
+
+    events.add(eventOne);
+    events.add(eventTwo);
+
+    CalendarGroup eventGroup = new CalendarGroup(events, startTime, endTime);
+    List<TimeRange> freeTimeRanges = eventGroup.getFreeTimeRanges();
+
+    Instant freeOneStart = startTime;
+    Instant freeOneEnd = eventOneStart;
+    Instant freeTwoStart = eventOneEnd;
+    Instant freeTwoEnd = eventTwoStart;
+    Instant freeThreeStart = eventTwoEnd;
+    Instant freeThreeEnd = endTime;
+
+    TimeRange freeOne = TimeRange.fromStartEnd(freeOneStart, freeOneEnd);
+    TimeRange freeTwo = TimeRange.fromStartEnd(freeTwoStart, freeTwoEnd);
+    TimeRange freeThree = TimeRange.fromStartEnd(freeThreeStart, freeThreeEnd);
+
+    List<TimeRange> freeTimeRangesAfterDelete = eventGroup.deleteFreeTimeRange(freeTwo);
+    Assert.assertTrue(freeTimeRangesAfterDelete.get(0).equals(freeOne));
+    Assert.assertTrue(freeTimeRangesAfterDelete.get(1).equals(freeThree));
+
+    // Creates a new time range that is 10 seconds shorter than free time One.
+    Instant freeOneEndNew = freeOneEnd.minusSeconds(10);
+    TimeRange freeOneNew = TimeRange.fromStartEnd(freeOneStart, freeOneEndNew);
+
+    List<TimeRange> freeTimeRangesAfterModify =
+        eventGroup.modifyFreeTimeRange(
+            /* originalFreeTimeRange= */ freeOne, /* newFreeTimeRange= */ freeOneNew);
+    Assert.assertTrue(freeTimeRangesAfterModify.get(0).end().equals(freeOneNew.end()));
```

Comment:
It seems like you can also use the collection comparison here if you'd like. Alternatively, we can probably use `Assert.assertEquals(freeOneNew.end(), freeTimeRangesAfterModify.get(0).end())`

Reply:
Updated.